api/v1beta1/common_openstackbaremetalset.go

```diff
@@ -123,17 +123,35 @@ func VerifyBaremetalSetScaleUp(log logr.Logger, instance *OpenStackBaremetalSet,
 	if newBmhsNeededCount > 0 {
 		// We have new replicas requested, so search for baremetalhosts that don't have consumerRef or Online set
 
+		labelStr := ""
+
+		if len(instance.Spec.BmhLabelSelector) > 0 {
+			labelStr = fmt.Sprintf(" with labels %v", instance.Spec.BmhLabelSelector)
+			labelStr = strings.Replace(labelStr, "map[", "[", 1)
+		}
+
+		log.Info(fmt.Sprintf("Attempting to find %d BaremetalHost(s)%s for scale-up of OpenStackBaremetalSet %s", newBmhsNeededCount, labelStr, instance.Name))
+
 		for _, baremetalHost := range allBmhs.Items {
-			if baremetalHost.Spec.Online || baremetalHost.Spec.ConsumerRef != nil {
-				continue
+			mismatch := false
+
+			if baremetalHost.Spec.Online {
+				log.Info(fmt.Sprintf("BaremetalHost %s cannot be used because it is already online", baremetalHost.ObjectMeta.Name))
+				mismatch = true
+			}
+
+			if baremetalHost.Spec.ConsumerRef != nil {
+				log.Info(fmt.Sprintf("BaremetalHost %s cannot be used because it already has a consumerRef", baremetalHost.ObjectMeta.Name))
+				mismatch = true
```

Comment:
wondering if it is ok to just continue instead of set mismatch?

Reply:
hrmm, yes think continuing here might be better. Otherwise would verifyBaremetalSetHardwareMatch would result in a lot of log spam for hosts that are already in use?

Reply:
I'll revisit this and try to find some balance.  I guess the question is what constitutes "more detail" and what veers into the category of "spam" as far as the RFE is concerned.